code/game/objects/items/devices/transfer_valve.dm

```diff
@@ -113,36 +118,40 @@
 
 /obj/item/device/transfer_valve/Topic(href, href_list)
 	..()
-	if( usr.stat || usr.restrained() )
+	if(usr.incapacitated())
 		return 0
-	if(src.loc != usr)
+	if(loc != usr)
 		return 0
 	if(tank_one && href_list["tankone"])
 		split_gases()
 		valve_open = 0
-		tank_one.loc = get_turf(src)
+		tank_one.forceMove(get_turf(src))
 		tank_one = null
 		update_icon()
+		if((!tank_two || tank_two.w_class < 4) && (w_class > 3))
+			w_class = 3
 	else if(tank_two && href_list["tanktwo"])
 		split_gases()
 		valve_open = 0
-		tank_two.loc = get_turf(src)
+		tank_two.forceMove(get_turf(src))
 		tank_two = null
 		update_icon()
+		if((!tank_one || tank_one.w_class < 4) && (w_class > 3))
+			w_class = 3
 	else if(href_list["open"])
 		toggle_valve()
 	else if(attached_device)
 		if(href_list["rem_device"])
-			attached_device.loc = get_turf(src)
-			attached_device:holder = null

```

Comment:
Reeee; why wasn't this typecast for an assembly when `attached_device` can *only* be an assembly?